apis/v1alpha2/validation/httproute_test.go

```diff
@@ -0,0 +1,323 @@
+										BackendRef: &gatewayv1a2.BackendObjectReference{
+											Name: testService,
+											Port: portNumberPtr(8080),
+										},
+									},
+								},
+								{
+									Type: gatewayv1a2.HTTPRouteFilterRequestHeaderModifier,
+									RequestHeaderModifier: &gatewayv1a2.HTTPRequestHeaderFilter{
+										Set: []gatewayv1a2.HTTPHeader{
+											{
+												Name:  "special-header",
+												Value: "foo",
+											},
+										},
+									},
+								},
+								{
+									Type: gatewayv1a2.HTTPRouteFilterRequestMirror,
+									RequestMirror: &gatewayv1a2.HTTPRequestMirrorFilter{
+										BackendRef: &gatewayv1a2.BackendObjectReference{
+											Name: testService,
+											Port: portNumberPtr(8080),
+										},
+									},
+								},
+								{
+									Type: gatewayv1a2.HTTPRouteFilterRequestHeaderModifier,
+									RequestHeaderModifier: &gatewayv1a2.HTTPRequestHeaderFilter{
+										Add: []gatewayv1a2.HTTPHeader{
+											{
+												Name:  "my-header",
+												Value: "bar",
+											},
+										},
+									},
+								},
+								{
+									Type: gatewayv1a2.HTTPRouteFilterRequestMirror,
+									RequestMirror: &gatewayv1a2.HTTPRequestMirrorFilter{
+										BackendRef: &gatewayv1a2.BackendObjectReference{
+											Name: specialService,
+											Port: portNumberPtr(8080),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			errCount: 2,
+		},
+		{
+			name: "valid httpRoute with duplicate ExtensionRef filters",
+			hRoute: gatewayv1a2.HTTPRoute{
+				Spec: gatewayv1a2.HTTPRouteSpec{
+					Rules: []gatewayv1a2.HTTPRouteRule{
+						{
+							Matches: []gatewayv1a2.HTTPRouteMatch{
+								{
+									Path: &gatewayv1a2.HTTPPathMatch{
+										Type:  pathMatchTypePtr("Prefix"),
+										Value: utilpointer.String("/"),
+									},
+								},
+							},
+							Filters: []gatewayv1a2.HTTPRouteFilter{
+								{
+									Type: gatewayv1a2.HTTPRouteFilterRequestHeaderModifier,
+									RequestHeaderModifier: &gatewayv1a2.HTTPRequestHeaderFilter{
+										Set: []gatewayv1a2.HTTPHeader{
+											{
+												Name:  "special-header",
+												Value: "foo",
+											},
+										},
+									},
+								},
+								{
+									Type: gatewayv1a2.HTTPRouteFilterRequestMirror,
+									RequestMirror: &gatewayv1a2.HTTPRequestMirrorFilter{
+										BackendRef: &gatewayv1a2.BackendObjectReference{
+											Name: testService,
+											Port: portNumberPtr(8080),
+										},
+									},
+								},
+								{
+									Type: "ExtensionRef",
+								},
+								{
+									Type: "ExtensionRef",
+								},
+								{
+									Type: "ExtensionRef",
+								},
+							},
+						},
+					},
+				},
+			},
+			errCount: 0,
+		},
+	}
+	for _, tt := range tests {
+		// copy variable to avoid scope problems with ranges
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			errs := ValidateHTTPRoute(&tt.hRoute)
```

Comment:
Something we have learned with Service validation/testing. If you are testing the global validation function then this test will become super complex as you add more to it. Instead of having one massive testing function let us have one test per validation-specialized-func in this case test `validateHTTPRouteUniqueFilters` not `ValidateHTTPRoute`

Reply:
The issue exist in all tests. But i will not add it for each test.